Crash when formatting emptied file

### Server Version

```
0.18.3
```

### Terraform Version

```
0.13.1
```

### Client Version

```
VS Code Terraform 2.13.0
```

### Log Output
```
panic: runtime error: index out of range [-1]

goroutine 92 [running]:
github.com/hashicorp/terraform-ls/internal/hcl.diffLines(0xc00a1ed12d, 0x7, 0xc009b4e290, 0x1, 0x1, 0xc00bdcf120, 0x2, 0x2, 0x1, 0xb245d00, ...)
	/Users/radeksimko/gopath/src/github.com/hashicorp/terraform-ls/internal/hcl/diff.go:116 +0x111a
github.com/hashicorp/terraform-ls/internal/hcl.Diff(0xb245d00, 0xc007398948, 0xc00021c200, 0x0, 0x200, 0xc000c6c600, 0x1, 0x600, 0x600, 0x0, ...)
	/Users/radeksimko/gopath/src/github.com/hashicorp/terraform-ls/internal/hcl/diff.go:47 +0x185
github.com/hashicorp/terraform-ls/internal/langserver/handlers.(*logHandler).TextDocumentFormatting(0xc0001307e8, 0x1940560, 0xc009c607e0, 0xc00a7aec80, 0x34, 0x4000000000000000, 0x1, 0x0, 0x0, 0x1940560, ...)
	/Users/radeksimko/gopath/src/github.com/hashicorp/terraform-ls/internal/langserver/handlers/formatting.go:44 +0x395
reflect.Value.call(0x15aa300, 0xc009ae5ee0, 0x13, 0x163fc46, 0x4, 0xc009c609c0, 0x2, 0x2, 0xc009c609c0, 0x199, ...)
	/Users/radeksimko/.goenv/versions/1.16/src/reflect/value.go:476 +0x8e7
reflect.Value.Call(0x15aa300, 0xc009ae5ee0, 0x13, 0xc009c609c0, 0x2, 0x2, 0x2, 0xc009c609c0, 0x1)
	/Users/radeksimko/.goenv/versions/1.16/src/reflect/value.go:337 +0xb9
github.com/creachadair/jrpc2/handler.newHandler.func8(0x1940560, 0xc009c607e0, 0xc000dd3ec0, 0x0, 0x0, 0x10, 0xc009ae5ee0)
	/Users/radeksimko/gopath/pkg/mod/github.com/creachadair/jrpc2@v0.17.0/handler/handler.go:212 +0x1f4
github.com/creachadair/jrpc2/handler.Func.Handle(...)
	/Users/radeksimko/gopath/pkg/mod/github.com/creachadair/jrpc2@v0.17.0/handler/handler.go:23
github.com/hashicorp/terraform-ls/internal/langserver/handlers.handle(0x1940560, 0xc009c607e0, 0xc000dd3ec0, 0x15aa300, 0xc009ae5ee0, 0x18a6950, 0x1940560, 0xc009c607e0, 0xc009c60701)
	/Users/radeksimko/gopath/src/github.com/hashicorp/terraform-ls/internal/langserver/handlers/service.go:423 +0x94
github.com/hashicorp/terraform-ls/internal/langserver/handlers.(*service).Assigner.func11(0x1940560, 0xc009c607b0, 0xc000dd3ec0, 0x0, 0x0, 0x0, 0x0)
	/Users/radeksimko/gopath/src/github.com/hashicorp/terraform-ls/internal/langserver/handlers/service.go:277 +0x252
reflect.Value.call(0x15c3360, 0xc00353dfb0, 0x13, 0x163fc46, 0x4, 0xc009c60750, 0x2, 0x2, 0xc009c60750, 0x0, ...)
	/Users/radeksimko/.goenv/versions/1.16/src/reflect/value.go:476 +0x8e7
reflect.Value.Call(0x15c3360, 0xc00353dfb0, 0x13, 0xc009c60750, 0x2, 0x2, 0x2, 0xc009c60750, 0x1)
	/Users/radeksimko/.goenv/versions/1.16/src/reflect/value.go:337 +0xb9
github.com/creachadair/jrpc2/handler.newHandler.func8(0x1940560, 0xc009c60720, 0xc000dd3ec0, 0x158d140, 0xc009ae5eb0, 0x15a96a0, 0xc007398840)
	/Users/radeksimko/gopath/pkg/mod/github.com/creachadair/jrpc2@v0.17.0/handler/handler.go:212 +0x1f4
github.com/creachadair/jrpc2/handler.Func.Handle(0xc003537d60, 0x1940560, 0xc009c60720, 0xc000dd3ec0, 0x0, 0x0, 0x1940560, 0xc009c60720)
	/Users/radeksimko/gopath/pkg/mod/github.com/creachadair/jrpc2@v0.17.0/handler/handler.go:23 +0x44
github.com/creachadair/jrpc2.(*Server).invoke(0xc00354c000, 0x19404b8, 0xc000dd3f00, 0x19369c0, 0xc003537d60, 0xc000dd3ec0, 0x0, 0x0, 0x0, 0x0, ...)
	/Users/radeksimko/gopath/pkg/mod/github.com/creachadair/jrpc2@v0.17.0/server.go:344 +0x1ab
github.com/creachadair/jrpc2.(*Server).dispatch.func1.1()
	/Users/radeksimko/gopath/pkg/mod/github.com/creachadair/jrpc2@v0.17.0/server.go:226 +0xb7
github.com/creachadair/jrpc2.(*Server).dispatch.func1(0x0, 0x0)
	/Users/radeksimko/gopath/pkg/mod/github.com/creachadair/jrpc2@v0.17.0/server.go:231 +0x16a
github.com/creachadair/jrpc2.(*Server).serve.func1(0xc00354c000, 0xc000b95b00)
	/Users/radeksimko/gopath/pkg/mod/github.com/creachadair/jrpc2@v0.17.0/server.go:152 +0x51
created by github.com/creachadair/jrpc2.(*Server).serve
	/Users/radeksimko/gopath/pkg/mod/github.com/creachadair/jrpc2@v0.17.0/server.go:150 +0x71
```

### Expected Behavior

LS `textDocument/formatting` method to return empty response when the document is empty.

### Actual Behavior

LS crashes

### Steps to Reproduce

 - create or open any non-empty file
 - remove all content
 - try saving it with auto-formatting enabled

--- a/internal/hcl/diff.go
+++ b/internal/hcl/diff.go
@@ -112,9 +112,25 @@ func diffLines(filename string, beforeLines, afterLines source.Lines) filesystem
 			}
 
 			if c.Tag == OpInsert {
-				insertRng := &hcl.Range{}
-				insertRng.Start = beforeLines[beforeStart-1].Range().End
-				insertRng.End = beforeLines[beforeStart-1].Range().End
+				insertRng := &hcl.Range{
+					Filename: filename,
+					Start:    hcl.InitialPos,
+					End:      hcl.InitialPos,
+				}
+
+				if beforeStart == beforeEnd {
+					line := beforeLines[beforeStart]
+					insertRng = line.Range().Ptr()
+				} else {
+					for i, line := range beforeLines[beforeStart:beforeEnd] {
+						if i == 0 {
+							insertRng = line.Range().Ptr()
+							continue
+						}
+						insertRng.End = line.Range().End
+					}
+				}
+
 				var newBytes []byte
 
 				for _, line := range afterLines[afterStart:afterEnd] {

--- a/internal/hcl/diff_test.go
+++ b/internal/hcl/diff_test.go
@@ -147,8 +147,9 @@ ccc`,
 				&fileChange{
 					newText: "\n",
 					rng: &hcl.Range{
-						Start: hcl.Pos{Line: 3, Column: 1, Byte: 30},
-						End:   hcl.Pos{Line: 3, Column: 1, Byte: 30},
+						Filename: "test.tf",
+						Start:    hcl.Pos{Line: 3, Column: 1, Byte: 30},
+						End:      hcl.Pos{Line: 3, Column: 1, Byte: 30},
 					},
 				},
 			},
@@ -173,6 +174,22 @@ ccc`,
 						Filename: "test.tf",
 						Start:    hcl.Pos{Line: 3, Column: 1, Byte: 44},
 						End:      hcl.Pos{Line: 4, Column: 1, Byte: 45},
+					},
+				},
+			},
+		},
+		{
+			"empty to newline",
+			``,
+			`
+`,
+			filesystem.DocumentChanges{
+				&fileChange{
+					newText: "\n",
+					rng: &hcl.Range{
+						Filename: "test.tf",
+						Start:    hcl.Pos{Line: 1, Column: 1, Byte: 0},
+						End:      hcl.Pos{Line: 1, Column: 1, Byte: 0},
 					},
 				},
 			},

